lib/rails_edge_test/dsl/edge.rb

```diff
@@ -74,6 +74,9 @@ def produce_json(ivar: nil)
       ActiveSupport::JSON::Encoding.escape_html_entities_in_json = false
       if ivar
         value = controller.send(:instance_variable_get, ivar)
+        JSON.pretty_unparse(value.as_json)
+      elsif response[1]['Content-Type']&.starts_with?('application/json')
+        value = JSON.parse(response[2].body)
         JSON.pretty_unparse(value)
       else
         response[2].body
```

Comment:
What are you imagining is the use case of this "else" case, and why are you not prettying it?

Reply:
else would be when the response body is not JSON

Reply:
(not needed by us at the moment, but maybe you might want to use rails_edge_test with an endpoint that returns XML or CSV or HTML or something?